### PR TITLE
[MGDOBR-6] Attempt to fix flaky test and improve Exception message for clarity

### PR DESCRIPTION
**Pull request by robpblake**

### CODE REVIEW

**Review by r00ta — APPROVED** [review]

File: manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java [hunk 1]
```diff
@@ -34,7 +34,7 @@
 public class ShardBridgesSyncAPI {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ShardBridgesSyncAPI.class);
-    private static final List<BridgeStatus> statuses = Arrays.asList(BridgeStatus.REQUESTED, BridgeStatus.PROVISIONING, BridgeStatus.DELETION_REQUESTED);
```
Comment (danielezonca) [inline manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java:37]: Isn't this change just hiding in all the endpoints bridges in `PROVISIONING` status? Why this change? Can't this happen in real world?

Comment (robpblake) [reply manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java:37]: Yep it can happen in the real-world, but not really in the current implementation given that everything is tied into a single deployment and we just start from fresh whenever we deploy the components. 

As/when we start to split things out, we support restarting of components and also look into correct failing handling, then we'll need to revisit what is returned from this API.

Comment (danielezonca) [reply manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java:37]: Thanks for the clarification, do we have a ticket for that? So that we keep this in the backlog

**Review by danielezonca — APPROVED** [review]
